adr/ADR-43.md

```diff
@@ -0,0 +1,102 @@
+|---------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| String idPrefix                             | used to make unique identifiers around each message. Defaults to a NUID                                                                                                                                                                                                      |
+| int maxInFlight                             | no more than this number of messages can be waiting for publish ack. Defaults to 50.                                                                                                                                                                                         |
+| int refillAllowedAt                         | if the queue size reaches maxInFlight, a hold is placed so no more messages can be published until the in flight queue contains this amount or less messages, at which time the hold is removed. Defaults to 0 which would be full sawtooth. Non zero provides for a window. |
+| RetryConfig retryConfig                     | if the user wants to publish with retries, they must supply a config, otherwise the publish will be attempted only once.                                                                                                                                                     |
+| long pollTime                               | the amount of time in ms to poll any given queue. Ensures polling doesn't block indefinitely. Defaults to 100ms                                                                                                                                                              |
+| long holdPauseTime                          | the amount of time in ms to pause between checks when hold is on. Defaults to 100ms                                                                                                                                                                                          |
+| long waitTimeout                            | the timeout when waiting for a publish to be acknowledged. Defaults to 5000ms                                                                                                                                                                                                |
+| PublisherListener publisherListener         | a callback for the user to see what's going on in the workflow, see description of Flight later                                                                                                                                                                              |
+
+
+## PublisherListener Interface
+
+The callback interface for the user to get information about the publish workflow
```

Comment:
The callback interface would not work in many languages as s good API.
This should be described in more general terms.

Reply:
Such as?